hoomd/md/module-md.cc

```diff
@@ -110,6 +110,7 @@ void export_TwoStepNVTAlchemy(pybind11::module& m);
 void export_FIREEnergyMinimizer(pybind11::module& m);
 void export_MuellerPlatheFlow(pybind11::module& m);
 void export_AlchemostatTwoStep(pybind11::module& m);
+void export_HalfStepHook(pybind11::module& m);
```

Comment:
Add `HalfStepHook.cc` to the `_md_sources` list in `md/CMakeLists.cc`. This, combined with the `PYBIND11_EXPORT` modification should fix the linker errors in the compiled unit tests.